src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -205,36 +205,38 @@ public static void Bind(this IConfiguration configuration, object instance, Acti
         }
 
         [RequiresUnreferencedCode(PropertyTrimmingWarningMessage)]
-        private static void BindNonScalar(this IConfiguration configuration, object instance, BinderOptions options)
+        private static void BindNonScalar(this IConfiguration configuration, object? instance, BinderOptions options)
         {
-            if (instance != null)
```

Comment:
Is this refactoring necessary? It makes reviewing the nullable annotations much harder.